core/shared/src/main/scala/zio/Accessible.scala

```diff
@@ -47,4 +47,9 @@ trait Accessible[R] {
     f: R => ZIO[R1, E, A]
   )(implicit tag: Tag[R], trace: ZTraceElement): ZIO[R1, E, A] =
     ZIO.serviceWithZIO[R](f)
+
+  def apply[A](
```

Comment:
Do we have tests for this? It is somewhat surprising to me that it works to have both of these methods since a `ZIO` is an `A`.

Reply:
I've added a few tests in code base. All works fine, except for Scala 2.11, because compiler [can't infer types](https://github.com/zio/zio/runs/5247029734?check_suite_focus=true#step:6:54) of accessing functions passed to `apply` properly if there are two overloaded `apply` methods. Unfortunately I couldn't find any meaningful way to solve that problem for 2.11, if it's possible at all.

So I left old functionality of `Accessible` for 2.11 and added the enhancement only for 2.12+. What do you think?

Reply:
There is a lot of value in having a consistent cross platform interface. I'm not sure the benefits here are worth deviating from that.